shared/constants/chat2/convostate.tsx

```diff
@@ -718,11 +722,24 @@ const createSlice: Z.ImmerStateCreator<ConvoState> = (set, get) => {
     })
   }
 
-  // used by loadMoreMessages
-  let scrollBackOldest = T.Chat.numberToOrdinal(-1)
-  let scrollBackOldestTime = 0
-  let scrollForwardNewest = T.Chat.numberToOrdinal(-1)
-  let scrollForwardNewestTime = 0
+  let lastScrollNumOrdinals = 0
+  let lastScrollTime = 0
+  const okToLoadMore = (n: number) => {
```

Comment:
the thread impls can make calls especially right as they're loading new things so its very easy to make 2x calls as you're scrolling up quickly. instead we keep the last num cached and a timeout value and constrain it. Watching scrolling w/ logging on this acts better